neuromation/cli/formatters/ftable.py

```diff
@@ -0,0 +1,120 @@
+from dataclasses import dataclass
+from enum import Enum
+from itertools import zip_longest
+from textwrap import wrap
+from typing import Any, Iterator, List, Optional, Sequence
+
+
+__all__ = ["table"]
+__version__ = "0.1"
+
+
+@dataclass(frozen=True)
+class ColumnWidth:
+    min: Optional[int] = None
+    max: Optional[int] = None
+    width: Optional[int] = None
+
+    def __post_init__(self) -> None:
+        if self.width and self.width < 0:
+            raise ValueError("Width must be positive integer")
+        if self.min and self.min < 0:
+            raise ValueError("Min must be positive integer")
+        if self.max and self.max < 0:
+            raise ValueError("Max must be positive integer")
+
+
+class Align(str, Enum):
+    CENTER = "center"
+    LEFT = "left"
+    RIGHT = "right"
+
+
+_align_to_format = {
+    Align.LEFT: str.ljust,
+    Align.RIGHT: str.rjust,
+    Align.CENTER: str.center,
+    None: str.ljust,
+}
+
+
+def table(
+    rows: Sequence[Sequence[str]],
+    widths: Sequence[ColumnWidth] = (),
+    aligns: Sequence[Align] = (),
+    max_width: Optional[int] = None,
+) -> Iterator[str]:
+
+    # Columns widths calculation
+    if len(widths) < len(rows[0]):
+        widths = tuple(widths) + tuple([ColumnWidth()]) * (len(rows[0]) - len(widths))
+    calc_widths: List[int] = []
+    for i, width in enumerate(widths):
+        max_cell_width: int = max(len(row[i]) for row in rows)
+        width_min = width.min if width.min else width.width
+        width_max = width.max if width.max else width.width
+
+        if (not width_min or width_min <= max_cell_width) and (
```

Comment:
If `width_min=3`, `width_max=5` and `max_cell_width=1` you will append 5.

Please add tests for `max_cell_width < width_min < width_max`, `width_min < max_cell_width < width_max` and `width_min < width_max < max_cell_width`.

I would write the code as:
```python
if width_min and width_min > max_cell_width:
    max_cell_width = width_min
if width_max and width_max < max_cell_width:
    max_cell_width = width_max
calc_widths.append(max_cell_width)
```
or
```python
max_cell_width = max(max_cell_width, width_min or max_cell_width)
max_cell_width = min(max_cell_width, width_max or max_cell_width)
calc_widths.append(max_cell_width)
```

Reply:
Thnx
Fixed, test cases added